### DISCUSSION TITLE
Is there any way to get the url of the document?

### DISCUSSION THREAD

**Discussion by wenscl** [diegomura/react-pdf#1155] [Q&A] [answered]
Hi! I would like to know if there is any way to get the url of the document to use it in the onclick event of the [menu item](https://material-ui.com/es/api/menu-item/#menuitem-api) of react mui.
I tried this but it doesn't work.

```
var foo = undefined;
<BlobProvider document={document}>
  {({ url }) => {
    foo = url;
  }}
</BlobProvider>;

<MenuItem onClick={() => window.open(foo, "_blank")}>
  Exportar a PDF
</MenuItem>
```

**Comment by diegomura** [comment, marked as answer]
That' not how the `BlobProvider` is supposed to work. Defining just variables inside a React component and expecting the rendering to have the right value is unreliable. You can wither try saving the `url` in the component state, or consume `BlobProvider` like this:

```js
<BlobProvider document={document}>
  <MenuItem onClick={() => window.open(url, "_blank")}>
    Exportar a PDF
  </MenuItem>
</BlobProvider>;
```
